python/pysnaptest/snapshot.py

```diff
@@ -83,11 +83,16 @@ def assert_snapshot(
 
 
 def insta_snapshot(
-    result: Any, snapshot_path: str | None = None, snapshot_name: str | None = None
+    result: Any, snapshot_path: str | None = None, snapshot_name: str | None = None, redactions: Optional[Dict[str, str]] = None
 ):
     if isinstance(result, dict) or isinstance(result, list):
-        assert_json_snapshot(result, snapshot_path, snapshot_name)
-    elif try_is_pandas_df(result) or try_is_polars_df(result):
+        assert_json_snapshot(result, snapshot_path, snapshot_name, redactions)
+        return
+
+    if redactions is not None:
+        raise ValueError("Redactions may only be used with json snapshots. Pass a list or dict instead.")
+    
+    if try_is_pandas_df(result) or try_is_polars_df(result):
```

Comment:
Would like to support this case, but it turns out we actually compare these as strings at the moment, and they do not run through the rust csv serialisation as a result.